src/lib/Libutil/pbs_aes_encrypt.c

```diff
@@ -33,27 +33,37 @@ extern unsigned char pbs_aes_iv[];
 int
 pbs_encrypt_data(char *uncrypted, int *credtype, size_t len, char **crypted, size_t *outlen)
 {
-        EVP_CIPHER_CTX ctx;
         int plen, len2 = 0;
         char *cblk;
 
-        EVP_CIPHER_CTX_init(&ctx);
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#define CIPHER_CONTEXT_INIT(v) EVP_CIPHER_CTX_init(v)
+#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_cleanup(v)
+        EVP_CIPHER_CTX real_ctx;
+        EVP_CIPHER_CTX *ctx = &real_ctx;
+#else
+#define CIPHER_CONTEXT_INIT(v) v = EVP_CIPHER_CTX_new(); if (!v) return -1;
+#define CIPHER_CONTEXT_CLEAN(v) EVP_CIPHER_CTX_free(v)
+        EVP_CIPHER_CTX *ctx = NULL;
+#endif
 
-        if (EVP_EncryptInit_ex(&ctx, EVP_aes_256_cbc(), NULL, (const unsigned char *) pbs_aes_key, (const unsigned char *) pbs_aes_iv) == 0)
+        CIPHER_CONTEXT_INIT(ctx);
+
+        if (EVP_EncryptInit_ex(ctx, EVP_aes_256_cbc(), NULL, (const unsigned char *) pbs_aes_key, (const unsigned char *) pbs_aes_iv) == 0)
                 return -1;
 
-        plen = len + EVP_CIPHER_CTX_block_size(&ctx) + 1;
+        plen = len + EVP_CIPHER_CTX_block_size(ctx) + 1;
         cblk = malloc(plen);
         if (!cblk)
                 return -1;
 
-        if (EVP_EncryptUpdate(&ctx, cblk, &plen, uncrypted, len) == 0)
+        if (EVP_EncryptUpdate(ctx, cblk, &plen, uncrypted, len) == 0)
                 return -1;
```

Comment:
I also added the CIPHER_CONTEXT_CLEAN before the returns. CIPHER_CONTEXT_INIT allocates ctx.